src/app/MessageDef/WriteResponse.cpp

```diff
@@ -0,0 +1,145 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+/**
+ *    @file
+ *      This file defines WriteResponse parser and builder in CHIP interaction model
+ *
+ */
+
+#include "WriteResponse.h"
+#include "MessageDefHelper.h"
+
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdio.h>
+
+using namespace chip;
+using namespace chip::TLV;
+
+namespace chip {
+namespace app {
+CHIP_ERROR WriteResponse::Parser::Init(const chip::TLV::TLVReader & aReader)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    // make a copy of the reader here
+    mReader.Init(aReader);
+
+    VerifyOrExit(chip::TLV::kTLVType_Structure == mReader.GetType(), err = CHIP_ERROR_WRONG_TLV_TYPE);
+
+    // This is just a dummy, as we're not going to exit this container ever
+    chip::TLV::TLVType OuterContainerType;
+    err = mReader.EnterContainer(OuterContainerType);
+
+exit:
+    ChipLogFunctError(err);
+
+    return err;
+}
+
+#if CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK
+CHIP_ERROR WriteResponse::Parser::CheckSchemaValidity() const
+{
+    CHIP_ERROR err           = CHIP_NO_ERROR;
+    uint16_t TagPresenceMask = 0;
+    chip::TLV::TLVReader reader;
+    StatusList::Parser statusList;
+    PRETTY_PRINT("WriteResponse =");
+    PRETTY_PRINT("{");
+
+    // make a copy of the reader
+    reader.Init(mReader);
+
+    while (CHIP_NO_ERROR == (err = reader.Next()))
+    {
+        const uint64_t tag = reader.GetTag();
+
+        if (chip::TLV::ContextTag(kCsTag_StatusList) == tag)
```

Comment:
What if an unexpected tag (other than the status list) is met. Should it generate an error?